### PR TITLE
Projects with path in common

### PR DESCRIPTION
**Pull request by tkrajina**
If a project named "aaa" is imported, then you import
another named "aaa-bbb" and then try to git-tf clone "aaa-bbb"
it failed saying that you must be in a local path where "aaa"
was.

Thats because:

/path/to/workspace/aaa
/path/to/workspace/aaa-bbb

have a common os.path.commonprefix. And every run of clone in
/path/to/workspace/aaa-bbb keeps complaining that you need to be
in /path/to/workspace/aaa.

### CODE REVIEW

**Comment by tkrajina** [comment]
PS. I didn't want to but pulled a change in .gitignore, too. I can remove it, but -- is it OK if it stays like it is? I added just a few temporary files generated from vim editor.

**Comment by nodirt** [comment]
Hello Tomo

First of all thanks for fixing an issue.
Currently I'm very busy with my job and preparation of a talk on
[defineClass](http://github.com/nodirt/defineClass) I'm going to give
on the weekend. BTW, you may interested in defineClass, it's a OOP +
AOP for JavaScript.

I'll review your pull request on the next week

**Comment by tkrajina** [comment]
No problem for me.

defineClass looks interesting definitely! I may use it for a new project I'm just starting. 

PS. Any chance your talk will be uploaded online?
PPS. I'll have another pull request / suggestion later.

File: clone.py [hunk 1]
```diff
@@ -40,11 +40,17 @@ def _checkDirectory(self):
         workfold = tf('workfold .')
         pwd = os.path.abspath('.')
         folderMaps = re.findall('^\s*(\$[^:]+): (\S+)$', workfold, re.M)
-        serverRoot, localRoot = [(s, l) for s, l in folderMaps if os.path.commonprefix((l, pwd)) == l][0]
```
Comment (nodirt) [inline clone.py:43]: Thanks for fixing this. Let me bring a minor improvement/simplification.

As far as I understand, we first get the list of candidates (`commonprefix`) and then choose the best match (`localRoot==pwd`). That's how I did it initially, as far as I remember.  Looks we need the candidates only to print them in the error case. That's kinda pessimistic behavior. Now I think that the code would be clearer if we try to find the best match (`pwd==localRoot`) from the start. Then, if we didn't find the exact match, we print the candidates. This way we will have no the for loop with a single iteration.

I'm not sure how to propose code changes to a proposed pull request (forking your fork?) so I'll write the code right here:

``` python
if any(l == pwd for (s, l) in folderMaps):
    return

print('You must be in a mapped local folder to use "git tf clone":')
for serverRoot, localRoot in folderMaps:
    if os.path.commonprefix((localRoot, pwd)) == localRoot:
        printIndented(localRoot)
fail()
```

Note: I didn't run/check this code.

**Comment by nodirt** [comment]
@tkrajina, will you change the code or should I do it?

**Comment by tkrajina** [comment]
Hey Nodir,

Sorry, but I cant check your code at the moment (I have no TFS repository now). I can try in a few days if you want, or you can proceed yourself. Your improvements looks good to me. If you can test it -- I'm OK with it.

tk

**Comment by nodirt** [comment]
I need these changes, so I'll do it myself tomorrow. AFAIU pull request mechanics I will have to discard your pull request and put the aforementioned piece of code myself.

**Comment by tkrajina** [comment]
No problem for me. Just close the pull request and make your changes. The important thing is that the bug is fixed :)

**Comment by nodirt** [comment]
Done: d05b635703773d036caf0f8e9243d09d6918463a

**Pull request closed without merging**